ambry-replication/src/test/java/com/github/ambry/replication/ReplicationTest.java

```diff
@@ -601,7 +602,18 @@ public void replicaFromStandbyToLeaderTest() throws Exception {
     StorageManager storageManager = managers.getFirst();
     MockReplicationManager replicationManager = (MockReplicationManager) managers.getSecond();
     PartitionId existingPartition = replicationManager.partitionToPartitionInfo.keySet().iterator().next();
+    String currentdatacenter =
+        storageManager.getReplica(existingPartition.toString()).getDataNodeId().getDatacenterName();
     mockHelixParticipant.onPartitionBecomeLeaderFromStandby(existingPartition.toPathString());
+    List<ReplicaId> peerLeaderReplicasInReplicationManager =
+        replicationManager.getPeerLeaderReplicasByPartition().get(existingPartition.toString());
+    List<ReplicaId> peerLeaderReplicasInClusterMap =
+        (List<ReplicaId>) existingPartition.getReplicaIdsByState(ReplicaState.LEADER, null)
+            .stream()
+            .filter(r -> !r.getDataNodeId().getDatacenterName().equals(currentdatacenter))
+            .collect(Collectors.toList());
+    assertThat("mismatch in leader peer replicas", peerLeaderReplicasInReplicationManager,
+        is(peerLeaderReplicasInClusterMap));
     storageManager.shutdown();
   }
```

Comment:
Since you also made changes in LEADER -> STANDBY transition in replication manager, could you also add a test for this case?  (in `replicaFromLeaderToStandbyTest`)